### ISSUE TITLE
QSAR toolbox server error with vega addins

### ISSUE THREAD

**Issue by CHAYAWAN** [marcabrus/TBVegaAddin#7]
Hey! I am facing the following error related to the QSAR toolbox server when I am trying to predict the chemical using vega add ins v 1.0.3 as well as v 1.0.5. I have installed all the prerequisites like postgre, .net (4.7.2) java (64bit) at default location (C drive). but I am getting the following error again and again. Please give me sugesstions and help me.
25 February 2021 22:45:09
========== ServerInfo START ==========
Server Mode: SingleUser
Listening on: net.tcp://localhost:8801/WcfRequestProcessor, Local IPC
Database: Toolboxv21R1@localhost
Service State: Opened
Sessions: 1/1
Documents: 1/1
Chemical lists: 4/4
Tree nodes: 335/420
Tasks: 7/39
Server Uptime: 00:02:03
Operating System: Microsoft Windows 10 Home
Platform: 64-bit
Processor: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
Installed Memory: 16 GB
Used Memory: 1,212 MB
========== ServerInfo END ==========
========== ErrorLog START ==========
[02/25/2021 22:44:48]	Warning	While calculate c1ccccc1 by model VEGA - Ready Biodegradability model (IRFMN): Unable to perform Applicability Domain check
[02/25/2021 22:44:39]	Error	System.NullReferenceException: Object reference not set to an instance of an object.
   at net.sf.jni4net.jni.JNIEnv.GetFieldID(Class clazz, String name, String sig)
   at net.sf.jni4net.Ref.Init()
   at net.sf.jni4net.Bridge.BindCore(JNIEnv env, BridgeSetup newSetup)
   at net.sf.jni4net.Bridge.CreateJVM()
   at net.sf.jni4net.Bridge.CreateJVM(BridgeSetup setup)
   at VegaAddins.Qsar.QsarAddin.RetrieveModelPreD(ITbBasket target, Dictionary`2 Modelinfo) in C:\Users\kosmu\Source\Repos\marcabrus\TBVegaAddin\Qsar\QsarAddin.cs:line 177
   at VegaAddins.Qsar.QsarAddin.IsRelevantToChemical(ITbBasket target, String& reason) in C:\Users\kosmu\Source\Repos\marcabrus\TBVegaAddin\Qsar\QsarAddin.cs:line 106
   at LMC.Toolbox.Server.ObjectManager.TbObjectFactoryWrappers.TbQsarWrapper.IsRelevantToChemical(IWorkBasket target)
[02/25/2021 22:44:39]	Error	net.sf.jni4net.jni.JNIException: Can't initialize proxy java.lang.System ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> java.lang.LinkageError: runFinalizersOnExit
   at net.sf.jni4net.jni.JNIEnv.ExceptionTest()
   at net.sf.jni4net.jni.JNIEnv.GetStaticMethodID(Class clazz, String name, String sig)
   at java.lang.System.InitJNI(JNIEnv __env, Class __class)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at net.sf.jni4net.utils.Registry.RegisterStaticAndMethods(RegistryRecord record, JNIEnv env)
   --- End of inner exception stack trace ---
   at net.sf.jni4net.utils.Registry.RegisterStaticAndMethods(RegistryRecord record, JNIEnv env)
   at net.sf.jni4net.utils.Registry.BindJvm(RegistryRecord record, ClassLoader classLoader, JNIEnv env)
   at net.sf.jni4net.utils.Registry.BindJvm(RegistryRecord record, JNIEnv env)
   at net.sf.jni4net.utils.Registry.RegisterType(Type type, Boolean bindJVM, JNIEnv env)
   at net.sf.jni4net.utils.Registry.RegisterAssembly(Assembly assembly, Boolean bindJVM)
   at net.sf.jni4net.Bridge.RegisterAssembly(Assembly assembly)
   at net.sf.jni4net.Bridge.BindCore(JNIEnv env, BridgeSetup newSetup)
   at net.sf.jni4net.Bridge.CreateJVM()
   at net.sf.jni4net.Bridge.CreateJVM(BridgeSetup setup)
   at VegaAddins.Qsar.QsarAddin.RetrieveModelPreD(ITbBasket target, Dictionary`2 Modelinfo) in C:\Users\kosmu\Source\Repos\marcabrus\TBVegaAddin\Qsar\QsarAddin.cs:line 177
   at VegaAddins.Qsar.QsarAddin.IsRelevantToChemical(ITbBasket target, String& reason) in C:\Users\kosmu\Source\Repos\marcabrus\TBVegaAddin\Qsar\QsarAddin.cs:line 106
   at LMC.Toolbox.Server.ObjectManager.TbObjectFactoryWrappers.TbQsarWrapper.IsRelevantToChemical(IWorkBasket target)
========== ErrorLog END ==========

**Comment by marcabrus** [comment]
You should have installed java 32 bit, and you should check if  the environment is set correctly. try to run java -version

**Comment by CHAYAWAN** [comment]
Thank you Cosimo for the suggestions. I tried with the JAVA 32 bit and set the environment the environmental variable to the JDK directory. it did not worked and sent the error that its not compatible with JAVA 32. I again reinstalled the JAVA 64 bit error. please look at the following error when I am running the QSAR tool box with 64 bit java
Reasons:
Exception(s) on server
Exception(s) on server
While executing session method: QueryTaskProgress
Parameters:
===
Parameter type=LMC.Toolbox.CS.Declarations.Session.SessionAPIParameters.GetInfo.SessionQueryTaskParameters
ServerMethodName=QueryTaskProgress
Created: 01:57:47.819
Delivered on server: N/A
QueryTaskName = PredictChemicalsByQsar
Target document Id = 2
Target list Id = 4
===

===
LMC.Toolbox.CS.Declarations.Session.SessionAPIParameters.Qsars.PredictChemicalsByQsarAPIResult
Server task Id=44
ServeActionCaption: Predict chemical(s) by QSAR
WorkingMs: 344
Logged exception from server:
System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at net.sf.jni4net.utils.Registry.ResolveNew(Type type)
   at net.sf.jni4net.utils.Registry.GetCLRRecord(Type iface)
   at net.sf.jni4net.utils.Convertor.FullJ2C[TRes](JNIEnv env, JniLocalHandle obj)
   at insilico.vega.vegadockcli.ChemicalinSet.getDataset(String par0, String par1)
   at VegaAddins.Qsar.QsarAddinDefinitions.GetSet(Dictionary`2 Modelinfo, TbScale ScaleDeclaration, String Set) in C:\Users\kosmu\Source\Repos\marcabrus\TBVegaAddin\Qsar\QsarAddinDefinitions.cs:line 94
   at VegaAddins.Qsar.TbQsarAddinFactory.TrainingSet(ITbWorkTask task) in C:\Users\kosmu\Source\Repos\marcabrus\TBVegaAddin\Qsar\TbQsarAddinFactory.cs:line 142
   at LMC.Toolbox.Server.ObjectManager.TbObjectFactoryWrappers.TbQsarWrapper.RegisterQsarChemSet(IToolboxQsarChemSetData existing, QsarChemicalSetType setType, Int32 statCount, Int32 statPropCount, IDomainTask domainTask)
   at LMC.Toolbox.Server.ObjectManager.TbObjectFactoryWrappers.TbQsarWrapper.InternalPrepareChemicalsSets(ITbWorkTask task)
   at LMC.Toolbox.Server.ObjectManager.TbObjectFactoryWrappers.TbQsarWrapper.GetTrainingSetData(IToolboxTask task)
   at LMC.Toolbox.Server.Domain.Prediction.ToolboxQsarPrediction.GetPredictionChemicals(IToolboxTask task)
   at LMC.Toolbox.Server.Domain.Tools.PredictionTools.UpdateBasicQsarPrediction(IDomainTask task, ToolboxQsarPrediction prediction, String qsarReferenceString, IStructure targetChemical, Double[] xPredictors)
   at LMC.Toolbox.Server.ObjectManager.TbObjectFactoryWrappers.TbQsarWrapper.Predict(IWorkBasket target, Nullable`1 confLevel)
   at LMC.Toolbox.Server.Actions.Qsars.PredictChemicalsByQsarAction.DoWork(IDomainTask myTask)
   at LMC.Toolbox.Server.Session.ToolboxAction.SafeDoWork()
===

**Comment by CHAYAWAN** [comment]
Caio Cosimo, I hope you are fine and doing well. can you please suggest me compatible version of java that you are using currently?

**Comment by marcabrus** [comment]
Hi chayawan, can you run java -version in command prompt? 1.8.xxx should be fine

**Comment by CHAYAWAN** [comment]
Thank you Cosimo. I have installed the same. command prompt returning me
java version "1.8.0_281"
Java(TM) SE Runtime Environment (build 1.8.0_281-b09)
Java HotSpot(TM) Client VM (build 25.281-b09, mixed mode). 
I compared it with other example on internet and I found that 'sharing' is missing after the 'mixed mode' in last line of the internet. Is it mandatory for proper working of JAVA    
But the error persist
